    @Test
    public void testIncrement() throws IOException {
        Assert.assertEquals(
                Short.valueOf((short) 4), KeysFetcher.FromPreviousIntegral.increment((short) 3));
        Assert.assertEquals(Integer.valueOf(11), KeysFetcher.FromPreviousIntegral.increment(10));
        Assert.assertEquals(Long.valueOf(21), KeysFetcher.FromPreviousIntegral.increment(20L));
        Assert.assertEquals(
                BigInteger.valueOf(31),
                KeysFetcher.FromPreviousIntegral.increment(BigInteger.valueOf(30)));
        Assert.assertEquals(
                BigDecimal.valueOf(41),
                KeysFetcher.FromPreviousIntegral.increment(BigDecimal.valueOf(40)));

        try {
            KeysFetcher.FromPreviousIntegral.increment("boom");
        } catch (IOException e) {
            // expected
        }
    }

